plugins/ssr/ssrequality.ml

```diff
@@ -399,15 +406,16 @@ let pirrel_rewrite ?(under=false) ?(map_redex=id_map_redex) pred rdx rdx_ty new_
       let c1' = Global.constant_of_delta_kn (Constant.canonical (Constant.make2 mp l')) in
       sigma, EConstr.of_constr (mkConst c1')
   in
-  let proof = EConstr.mkApp (elim, [| rdx_ty; new_rdx; pred; p; rdx; c |]) in
+  let proof = EConstr.mkApp (elim, [| rdx_ty; new_rdx; pred |]) in
```

Comment:
Which one takes time? The CI failure seems because `p` is not there...

Reply:
Good question. We have that `p` is an evar created with the right type so it's a matter of checking that it is indeed well-typed. It's not clear what the invariants around pattern selection regarding typing are, so maybe we should be slightly more specified.

Reply:
It is not exactly the case, there is a subst1 and a beta in the type of p.
Maybe this is also why typing the application with it may need more conversion.

Maybe you could try creating p with the right, trivial to type check, type and only later beta reduce it (the please the eye of the user)?

Reply:
The error in the minimized example looks like a variable capture. It's a nonsensical type error, not a missing universe constraint.

Reply:
It seems that `rdx` is nonsense indeed, it's referring to an unbound rel. Are the pattern abstraction functions supposed to always produce a well-typed term (and thus is this a bug there) or should I always type-check the result?

Reply:
It is not a capture, the term reduced and the unbound variable disappears. This is why pattern selection, unification, did succeed in the first place even if you are replacing a term under a binder with an apparent, but non essential, use of the bound variable.

Reply:
something magic was going on anyway....

Reply:
Erm, but then what's the desired behaviour there?